tycho-its/src/test/java/org/eclipse/tycho/test/resolver/ResolverTests.java

```diff
@@ -104,4 +104,12 @@ public void testConsiderResolutionWithUsesDirectiveIfVanillaResolutionFails() th
 		verifier.executeGoal("compile");
 		verifier.verifyErrorFreeLog();
 	}
+
+	@Test
+	public void testResolveRebundledMavenDependencies() throws Exception {
+		Verifier verifier = getVerifier("target.maven.wrapAsBundle", true);
+		verifier.executeGoal("dependency:tree");
+		verifier.verifyErrorFreeLog();
+		verifier.verifyTextInLog("com.squareup.okhttp3:okhttp:jar:4.10.0:compile");
```

Comment:
As explained, `dependency:tree` will most likely never show wrapped bundles as their original ones. but we can have this test with `@Ignore` explaining that this is a missing feature, actually it would be great if Tycho could delay the wrapping part further down in the build and e.g. only perform it when the item is packed into a product, updatesite or used in a test.

Reply:
That sounds like a good solution. The CycloneDX plugin, for example, requires a valid list of dependencies to generate a bill-of-materials. Currently, those only work in a limited capacity.